src/pages/home/subPages/dashboard/hooks/useSelectedTagMemosManager.ts

```diff
@@ -1,70 +1,168 @@
 import { useEffect, useCallback, useState, useMemo } from 'react';
-import { forkJoin, from, lastValueFrom, map } from 'rxjs';
-import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useInfiniteQuery, useQueryClient } from '@tanstack/react-query';
 import { Memo, Tag } from 'pages/home/subPages/interfaces';
+import { SortOption } from 'pages/home/subPages/dashboard/interfaces';
 import * as Api from 'api';
 
+const TAG_LIMIT = 10;
+const MEMO_LIMIT = 10;
+
 const useSelectedTagMemosManager = (
-  tags: Tag[] | null,
-  selectedTag: Tag | null
+  selectedTag: Tag | null,
+  sortOption: SortOption
 ) => {
   const queryClient = useQueryClient();
   const [memoSectionListByTag, setMemoSectionListByTag] = useState<
-    { tag: Tag; childTags: Tag[] | null; memos: Memo[] }[]
+    { tag: Tag | null; childTags: Tag[] | null; memos: Memo[] }[]
   >([]);
+  const [tags, setTags] = useState<Tag[]>([]);
+  const [childTagPages, setChildTagPages] = useState<
+    Record<string, { currentPage: number; totalPage: number } | undefined>
+  >({});
+  const [rootTag, setRootTag] = useState<Tag>();
 
-  const { data: fetchedMemos = [], refetch } = useQuery({
+  const {
+    data: fetchedMemos,
+    fetchNextPage,
+    refetch,
+  } = useInfiniteQuery<Api.paginationDashboardResponse, Error>({
     queryKey: useMemo(
-      () => ['memos', tags?.map((tag) => tag.id) || selectedTag?.id],
-      [tags, selectedTag]
+      () => ['memos', selectedTag?.id, MEMO_LIMIT, sortOption],
+      [selectedTag, MEMO_LIMIT, sortOption]
     ),
-    queryFn: async () => {
-      if (tags?.length) {
-        return lastValueFrom(forkJoin(tags.map(fetchTagData)));
-      } else if (selectedTag) {
-        return lastValueFrom(
-          fetchMemosByTag(selectedTag.id).pipe(
-            map((memos) => [{ tag: selectedTag, childTags: null, memos }])
-          )
-        );
+    queryFn: async ({ pageParam = 1 }: any) => {
+      const response =
+        selectedTag !== null
+          ? await Api.getDashboardDataByTag({
+              parentTagId: selectedTag.id,
+              tagPage: pageParam,
+              tagLimit: TAG_LIMIT,
+              memoLimit: MEMO_LIMIT,
+              sortOrder: sortOption,
+            })
+          : await Api.getDashboardDataByTag({
+              tagPage: pageParam,
+              tagLimit: TAG_LIMIT,
+              memoLimit: MEMO_LIMIT,
+              sortOrder: sortOption,
+            });
```

Comment:
이러면 이제 Api.getDashboardDataByTag에서 parentTagId가 null인지 확인하니까 됐습니다.

```suggestion
      const response =await Api.getDashboardDataByTag({
              parentTagId: selectedTag.id,
              tagPage: pageParam,
              tagLimit: TAG_LIMIT,
              memoLimit: MEMO_LIMIT,
              sortOrder: sortOption,
            })
```